skxray/correlation.py

```diff
@@ -119,9 +119,11 @@ def multi_tau_auto_corr(num_levels, num_bufs, labels, images):
         raise ValueError("number of channels(number of buffers) in "
                          "multiple-taus (must be even)")
 
-    if labels.shape != images.operands[0].shape[1:]:
-        raise ValueError("Shape of the image stack should be equal to"
-                         " shape of the labels array")
+    if hasattr(images, 'frame_shape'):
+        # Give a user-friendly error if we can detect the shape from pims.
+        if labels.shape != images.frame_shape:
+            raise ValueError("Shape of the image stack should be equal to"
+                             " shape of the labels array")
```

Comment:
<rabbit hole> this should have an `elif` looking for `shape` and then use a `tee` to check the shape of the first image </rabbit hole>